kairosdb/src/main/scala/akka/stream/alpakka/kairosdb/KairosDBSinkStage.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.kairosdb
+
+import akka.Done
+import akka.stream.stage._
+import akka.stream.{Attributes, Inlet, SinkShape}
+import org.kairosdb.client.Client
+import org.kairosdb.client.builder.MetricBuilder
+import org.kairosdb.client.response.Response
+
+import scala.concurrent._
+import scala.util.{Failure, Success}
+
+final case class KairosSinkSettings(parallelism: Int) {
+  require(parallelism > 0)
+
+}
+
+object KairosSinkSettings {
+  val Defaults = KairosSinkSettings(1)
+}
+
+private object NullExecutionContext extends ExecutionContext {
+  override def execute(runnable: Runnable): Unit = ???
+
+  override def reportFailure(cause: Throwable): Unit = ???
+}
+
+class KairosDBSinkStage(settings: KairosSinkSettings,
+                        kairosClient: Client)(implicit executionContext: ExecutionContext = NullExecutionContext)
+    extends GraphStageWithMaterializedValue[SinkShape[MetricBuilder], Future[Done]] {
+
+  val in: Inlet[MetricBuilder] = Inlet("KairosSink.in")
+
+  override val shape: SinkShape[MetricBuilder] = SinkShape(in)
+
+  override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, Future[Done]) = {
+    val promise = Promise[Done]()
+    val logic = new KairosSinkStageLogic(in, shape, kairosClient, promise, settings)(executionContext)
+
+    (logic, promise.future)
+  }
+}
+
+private[kairosdb] class KairosSinkStageLogic(
+    in: Inlet[MetricBuilder],
+    shape: SinkShape[MetricBuilder],
+    kairosClient: Client,
+    promise: Promise[Done],
+    settings: KairosSinkSettings
+)(executionContext: ExecutionContext)
+    extends GraphStageLogic(shape)
+    with StageLogging {
+
```

Comment:
_Let me first fix my wording. I mean removing the **default** value for the implicit ec but keeping it implicit._

The IronMQ client is async using callbacks, there is no blocking{...} in there so this should not be an issue. I wonder if the internal ec from akka streams is set up to grow when there are blocking calls (like the global one).
Anyway, when implicit without default the user can still decide to provide mat.executionContext.

```scala
class KairosDBSinkStage(settings: KairosSinkSettings, kairosClient: Client)(implicit executionContext: ExecutionContext)
```